pytorch_lightning/metrics/functional/ir_average_precision.py

```diff
@@ -0,0 +1,53 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+
+
+def retrieval_average_precision(
+    preds: torch.Tensor,
+    target: torch.Tensor
+) -> torch.Tensor:
+    r"""
+    Computes average precision (for information retrieval), as explained
+    `here <https://en.wikipedia.org/wiki/Evaluation_measures_(information_retrieval)#Average_precision>`_.
+
+    `preds` and `target` should be of the same shape and live on the same device. If no `target` is ``True``,
```

Comment:
What happens if they user provides something else than True, False

Reply:
Works the same with integers like `target = torch.tensor([1, 0, 1])` and also with floats.

Reply:
so lets write that the case has to be binary...? and add check the inputs are valid

Reply:
Implemented the check and allowed only int and booleans for the `target`s.